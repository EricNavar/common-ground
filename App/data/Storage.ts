import AsyncStorage from '@react-native-async-storage/async-storage';

export const Storage = {
    get: async (key: string) : Promise<string | null> => {
        return await AsyncStorage.getItem(key);
    },
    set: async (key: string, value: string) => {
        await AsyncStorage.setItem(key, value).then(() => {
            console.log('Stored ', value, ' at ', key);
        }).catch((error: any) => {
            console.log('Error storing data at ', key, error);
        });
    },
};