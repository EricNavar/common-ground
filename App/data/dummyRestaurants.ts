export const losPollosHermanos = {
    name: 'Los Pollos Hermanos',
    pic: 'https://static.independent.co.uk/s3fs-public/thumbnails/image/2015/05/01/15/lospolloshermanos.jpg?width=1200',
};

export const tacoSal = {
    name: 'Taco Sal',
    pic: 'https://static.wikia.nocookie.net/breakingbad/images/c/c0/TacoSal.png/revision/latest?cb=20191120130106',
};


export const loyolasFamilyRestaurant = {
    name: 'Loyola\'s Family Restaurant',
    pic: 'https://www.breakingbad-locations.com/wp-content/uploads/2014/09/E06-Cornered.mkv_000921613.jpg',
};
