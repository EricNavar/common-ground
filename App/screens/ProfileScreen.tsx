import React from "react";
import {
  ScrollView,
  TouchableOpacity,
  Image,
  View,
  Text,
  StyleSheet,
  Button,
} from "react-native";
import { Contact, Profile } from "../commonTypes";
import { Chip } from "../components/Chip";
import { ContactBubble } from "../components/ContactBubble";
import { RestaurantBubble } from "../components/RestaurantBubble";
import { mapContactToProfile } from "../utils";

const styles = StyleSheet.create({
  profile: {
    paddingLeft: 20,
    paddingRight: 20,
    paddingBottom: 20,
    backgroundColor: "white",
    height: "100%",
  },
  settings: {
    width: 32,
    height: 32,
    paddings: "100%",
    position: "absolute",
    top: 0,
    right: 0,
  },
  image: {
    width: 150,
    height: 150,
    borderRadius: 75,
    marginLeft: "auto",
    marginRight: "auto",
  },
  center: {
    marginTop: 20,
    marginBottom: 20,
    justifyContent: "center",
  },
  name: {
    marginLeft: "auto",
    marginRight: "auto",
    marginTop: 15,
    fontSize: 24,
    fontWeight: "bold",
  },
  row: {
    display: "flex",
    flexDirection: "row",
    alignItems: "baseline",
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: "bold",
    marginTop: 30,
    marginBottom: 12,
    marginRight: "auto",
  },
  chipContainer: {
    display: "flex",
    flexDirection: "row",
  },
  bubble: {
    marginRight: 10,
  },
});

type ProfilePageProps = {
  navigation: {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    navigate: any;
  };
  profileData: Profile;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  route: any;
};

function ProfileScreen(props: ProfilePageProps) {
  const selfId = "4";

  // this is true if the user is looking at their own profile
  const myProfile = props.route.params.profileData.id === selfId;

  const onPressRestaurant = () => {
    props.navigation.navigate("Restaurant");
  };

  const onPressContact = (contact: Contact) => {
    props.navigation.navigate("Profile", {
      profileData: mapContactToProfile(contact),
    });
  };

  const onPressSettings = () => {
    props.navigation.navigate("Settings");
  };

  const onPressEditPreferences = () => {
    props.navigation.navigate("Preferences");
  };

  const onPressSeeAllPicks = () => {
    props.navigation.navigate("Restaurant List");
  };

  const onPressSeeAllContacts = () => {
    props.navigation.navigate("All Contacts");
  };

  if (!props.route.params.profileData) return <></>;

  return (
    <ScrollView style={styles.profile}>
      {myProfile && (
        <View style={styles.center}>
          <TouchableOpacity onPress={onPressSettings}>
            <Image
              source={require("../assets/settings.png")}
              style={styles.settings}
            />
          </TouchableOpacity>
        </View>
      )}
      <View style={styles.center}>
        <Image
          source={{ uri: props.route.params.profileData.profilePic }}
          style={styles.image}
        />
        <Text style={styles.name}>{props.route.params.profileData.name}</Text>
      </View>
      <View style={styles.row}>
        <Text style={styles.sectionTitle}>Preferences</Text>
        <Button onPress={onPressEditPreferences} title="Edit" />
      </View>
      <View style={styles.chipContainer}>
        {props.route.params.profileData.preferences.map((preference, index) => (
          <Chip text={preference} key={index} />
        ))}
      </View>
      <View style={styles.row}>
        <Text style={styles.sectionTitle}>Past picks</Text>
        <Button onPress={onPressSeeAllPicks} title="See all" />
      </View>
      <View style={styles.row}>
        {props.route.params.profileData.pastPicks.map((restaurant, index) => (
          <RestaurantBubble
            {...restaurant}
            onPress={onPressRestaurant}
            key={index}
            style={styles.bubble}
          />
        ))}
      </View>
      {myProfile && (
        <>
          <View style={styles.row}>
            <Text style={styles.sectionTitle}>Your contacts</Text>
            <Button onPress={onPressSeeAllContacts} title="See all" />
          </View>
          <View style={styles.row}>
            {props.route.params.profileData.recentContacts.map(
              (contact, index) => (
                <ContactBubble
                  {...contact}
                  onPress={() => onPressContact(contact)}
                  key={index}
                />
              )
            )}
          </View>
        </>
      )}
      <View style={{ height: 20 }} />
    </ScrollView>
  );
}

export { ProfileScreen };
