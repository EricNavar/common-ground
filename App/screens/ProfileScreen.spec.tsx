import { saulProfile, walterProfile } from '../data/dummyUsers';
import {ProfileScreen} from './ProfileScreen';
import {render} from '@testing-library/react-native';
import React from 'react';

describe('ProfileScreen', () => {
    const dummyNavigation = {
        navigate: '',
    };

    it('Renders user\'s own profile correctly', () => {
        const route = {
            params: {
                profileData: saulProfile,
            },
        };
        const result = render(<ProfileScreen profileData={saulProfile} navigation={dummyNavigation} route={route} />);

        expect(result.toJSON()).toMatchSnapshot();
    });

    it('Renders user\'s own profile correctly', () => {
        const route = {
            params: {
                profileData: walterProfile,
            },
        };
        const result = render(<ProfileScreen profileData={walterProfile} navigation={dummyNavigation} route={route} />);

        expect(result.toJSON()).toMatchSnapshot();
    });
});
