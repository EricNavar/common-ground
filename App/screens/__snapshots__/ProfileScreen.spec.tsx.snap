// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen Renders someone else's profile correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "white",
      "height": "100%",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <View>
    <View
      style={
        {
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQS7_zsiM5wfE2uac1B7Xk-jdVGpCKzrQqEwOaK9ak60O1iI4Don3BcWR_sOsumMQetlSw&usqp=CAU",
          }
        }
        style={
          {
            "borderRadius": 75,
            "height": 150,
            "marginLeft": "auto",
            "marginRight": "auto",
            "width": 150,
          }
        }
      />
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 15,
          }
        }
      >
        Walter
         
        White
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 12,
            "marginRight": "auto",
            "marginTop": 30,
          }
        }
      >
        Preferences
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ff6d6e",
            "borderRadius": 25,
            "height": 30,
            "marginRight": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          BBQ
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ff6d6e",
            "borderRadius": 25,
            "height": 30,
            "marginRight": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          $-$$
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ff6d6e",
            "borderRadius": 25,
            "height": 30,
            "marginRight": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          American
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ff6d6e",
            "borderRadius": 25,
            "height": 30,
            "marginRight": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          Mexican
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 12,
            "marginRight": "auto",
            "marginTop": 30,
          }
        }
      >
        Past picks
      </Text>
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                {
                  "color": "#ff6e6e",
                },
              ]
            }
          >
            See all
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://static.independent.co.uk/s3fs-public/thumbnails/image/2015/05/01/15/lospolloshermanos.jpg?width=1200",
            }
          }
          style={
            {
              "borderRadius": 8,
              "height": 100,
              "width": 100,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
              "overflow": "hidden",
              "width": 100,
            }
          }
        >
          Los Pollos Hermanos
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://static.wikia.nocookie.net/breakingbad/images/c/c0/TacoSal.png/revision/latest?cb=20191120130106",
            }
          }
          style={
            {
              "borderRadius": 8,
              "height": 100,
              "width": 100,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
              "overflow": "hidden",
              "width": 100,
            }
          }
        >
          Taco Sal
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://www.breakingbad-locations.com/wp-content/uploads/2014/09/E06-Cornered.mkv_000921613.jpg",
            }
          }
          style={
            {
              "borderRadius": 8,
              "height": 100,
              "width": 100,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
              "overflow": "hidden",
              "width": 100,
            }
          }
        >
          Loyola's Family Restaurant
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "height": 20,
        }
      }
    />
  </View>
</RCTScrollView>
`;

exports[`ProfileScreen Renders user's own profile correctly 1`] = `
<RCTScrollView
  style={
    {
      "backgroundColor": "white",
      "height": "100%",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
    }
  }
>
  <View>
    <View
      style={
        {
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "testUri": "../../../assets/settings.png",
            }
          }
          style={
            {
              "height": 32,
              "paddings": "100%",
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 32,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "justifyContent": "center",
          "marginBottom": 20,
          "marginTop": 20,
        }
      }
    >
      <Image
        source={
          {
            "uri": "https://external-preview.redd.it/Af65FrXSGlwOobPUnC3MNOI1D3xCCERFWoVFfhzpgio.png?format=pjpg&auto=webp&s=ed7b1feb0f8356ba51ea70f5b26b81c1868bc005",
          }
        }
        style={
          {
            "borderRadius": 75,
            "height": 150,
            "marginLeft": "auto",
            "marginRight": "auto",
            "width": 150,
          }
        }
      />
      <Text
        style={
          {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 15,
          }
        }
      >
        Saul
         
        Goodman
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 12,
            "marginRight": "auto",
            "marginTop": 30,
          }
        }
      >
        Preferences
      </Text>
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                {
                  "color": "#ff6e6e",
                },
              ]
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ff6d6e",
            "borderRadius": 25,
            "height": 30,
            "marginRight": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          BBQ
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ff6d6e",
            "borderRadius": 25,
            "height": 30,
            "marginRight": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          $-$$
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ff6d6e",
            "borderRadius": 25,
            "height": 30,
            "marginRight": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          American
        </Text>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ff6d6e",
            "borderRadius": 25,
            "height": 30,
            "marginRight": 8,
            "padding": 8,
          }
        }
      >
        <Text
          style={
            {
              "color": "white",
            }
          }
        >
          Mexican
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 12,
            "marginRight": "auto",
            "marginTop": 30,
          }
        }
      >
        Past picks
      </Text>
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                {
                  "color": "#ff6e6e",
                },
              ]
            }
          >
            See all
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://static.independent.co.uk/s3fs-public/thumbnails/image/2015/05/01/15/lospolloshermanos.jpg?width=1200",
            }
          }
          style={
            {
              "borderRadius": 8,
              "height": 100,
              "width": 100,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
              "overflow": "hidden",
              "width": 100,
            }
          }
        >
          Los Pollos Hermanos
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://static.wikia.nocookie.net/breakingbad/images/c/c0/TacoSal.png/revision/latest?cb=20191120130106",
            }
          }
          style={
            {
              "borderRadius": 8,
              "height": 100,
              "width": 100,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
              "overflow": "hidden",
              "width": 100,
            }
          }
        >
          Taco Sal
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 10,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://www.breakingbad-locations.com/wp-content/uploads/2014/09/E06-Cornered.mkv_000921613.jpg",
            }
          }
          style={
            {
              "borderRadius": 8,
              "height": 100,
              "width": 100,
            }
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
              "overflow": "hidden",
              "width": 100,
            }
          }
        >
          Loyola's Family Restaurant
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 12,
            "marginRight": "auto",
            "marginTop": 30,
          }
        }
      >
        Your contacts
      </Text>
      <View
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                {
                  "color": "#ff6e6e",
                },
              ]
            }
          >
            See all
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        {
          "alignItems": "baseline",
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 16,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://compote.slate.com/images/fb69a16d-7f35-4103-98c1-62d466196b9a.jpg?width=1200",
            }
          }
          style={
            [
              {
                "borderRadius": 42.5,
                "height": 85,
                "width": 85,
              },
              null,
            ]
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
            }
          }
        >
          Finger
           
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 16,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://static.wikia.nocookie.net/breakingbad/images/5/54/BCS_S3_GusFring.jpg/revision/latest?cb=20170327185354",
            }
          }
          style={
            [
              {
                "borderRadius": 42.5,
                "height": 85,
                "width": 85,
              },
              null,
            ]
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
            }
          }
        >
          Gus
           
          Fring
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 16,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQS7_zsiM5wfE2uac1B7Xk-jdVGpCKzrQqEwOaK9ak60O1iI4Don3BcWR_sOsumMQetlSw&usqp=CAU",
            }
          }
          style={
            [
              {
                "borderRadius": 42.5,
                "height": 85,
                "width": 85,
              },
              null,
            ]
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
            }
          }
        >
          Walter
           
          White
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 16,
            "opacity": 1,
          }
        }
      >
        <Image
          source={
            {
              "uri": "https://static.wikia.nocookie.net/villains/images/3/38/740full-stan-edgar.jpg/revision/latest/scale-to-width-down/250?cb=20190925225050",
            }
          }
          style={
            [
              {
                "borderRadius": 42.5,
                "height": 85,
                "width": 85,
              },
              null,
            ]
          }
        />
        <Text
          style={
            {
              "marginLeft": "auto",
              "marginRight": "auto",
              "marginTop": 8,
            }
          }
        >
          Stan
           
          Edgar
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "height": 20,
        }
      }
    />
  </View>
</RCTScrollView>
`;
